File: first_project/scripts/my_first_node.py
#!/usr/bin/env python2
import rospy

if __name__ == '__main__':
    rospy.init_node("test_node")

    rospy.loginfo("Hello from test node")
    rospy.logwarn("This is a warning")
    rospy.logerr("This is an error")

    rospy.sleep(1.0)

    rospy.loginfo("This is the end of the program")

File: src/my_first_node.cpp
#include "ros/ros.h"

int main (int argc, char **argv) {
    // Initiate a new ROS node named "listener"
	ros::init(argc, argv, "test_node");
    //create a node handle: it is reference assigned to a new node
	ros::NodeHandle nh;


    ROS_INFO("Hello from test node");
    ROS_WARN("This is a warning");
    ROS_ERROR("This is an error");

    ros::Duration(1.0).sleep(); // Sleep for the rest of the cycle, to enforce the loop rate

    ROS_INFO("This is the end of the program");
}